.github/workflows/dependency_review.yaml

```diff
@@ -0,0 +1,39 @@
+# See https://github.com/actions/dependency-review-action
+name: 'Dependency Review'
+on: [pull_request]
+permissions:
+  contents: read
+jobs:
+  dependency-review:
+    name: Security
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Dependency Security Review
+        uses: actions/dependency-review-action@v2
+        with:
+          fail-on-severity: low
+          allow-licenses: MIT
+  dependency-updates:
+    name: Versions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Dependency Versions Review
+        run: |
+          # error off since we use 'git diff' to determine changes
+          set +o errexit
+          go get -u all; go mod tidy
+          set -o errexit
+          TMP_GIT_STATUS_GO_MOD=$(git diff go.mod)
+          echo "TMP_GIT_STATUS_GO_MOD=${TMP_GIT_STATUS_GO_MOD}"
+
+          if [ "${TMP_GIT_STATUS_GO_MOD}" == "" ]; then
+            echo "Dependencies are NOT outdated"
+
+            else
+              echo "Dependencies are outdated"
+              exit 1
```

Comment:
could create a [warning annotation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message) in `go.mod` when deps are outdated but still `exit 0` from the action

Reply:
Of limited use, see wip 14

Reply:
it looks like annotations can be made to show up on a specific line in a file: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks#checks

Reply:
https://docs.github.com/en/rest/checks/runs#create-a-check-run